vividus-plugin-applitools/src/main/java/org/vividus/visual/eyes/service/ImageVisualTestingService.java

```diff
@@ -0,0 +1,119 @@
+import org.vividus.http.client.HttpResponse;
+import org.vividus.http.client.IHttpClient;
+import org.vividus.visual.eyes.factory.ImageEyesFactory;
+import org.vividus.visual.eyes.model.ApplitoolsVisualCheck;
+import org.vividus.visual.eyes.model.ApplitoolsVisualCheckResult;
+import org.vividus.visual.screenshot.ScreenshotProvider;
+
+@Named
+public class ImageVisualTestingService implements VisualTestingService
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(ImageVisualTestingService.class);
+
+    private final ImageEyesFactory eyesFactory;
+    private final ScreenshotProvider screenshotProvider;
+    private final IHttpClient httpClient;
+
+    public ImageVisualTestingService(ImageEyesFactory eyesFactory, ScreenshotProvider screenshotProvider,
+            IHttpClient httpClient)
+    {
+        this.eyesFactory = eyesFactory;
+        this.screenshotProvider = screenshotProvider;
+        this.httpClient = httpClient;
+    }
+
+    @Override
+    public ApplitoolsVisualCheckResult run(ApplitoolsVisualCheck applitoolsVisualCheck)
+    {
+        Eyes eyes = eyesFactory.createEyes(applitoolsVisualCheck);
+        TestResults testResults = null;
+        try
+        {
+            eyes.open(applitoolsVisualCheck.getAppName(), applitoolsVisualCheck.getBaselineName());
+            eyes.checkImage(screenshotProvider.take(applitoolsVisualCheck).getImage());
+        }
+        finally
+        {
+            testResults = eyes.close(false);
+        }
+        return createVisualCheckResult(testResults, applitoolsVisualCheck);
+    }
+
+    private ApplitoolsVisualCheckResult createVisualCheckResult(TestResults testResults,
+            ApplitoolsVisualCheck applitoolsVisualCheck)
+    {
+        ApplitoolsVisualCheckResult visualCheckResult = new ApplitoolsVisualCheckResult(applitoolsVisualCheck);
+        visualCheckResult.setBatchUrl(testResults.getUrl());
+        visualCheckResult.setPassed(testResults.isPassed());
+        StepInfo stepInfo = testResults.getStepsInfo()[0];
+        visualCheckResult.setStepUrl(stepInfo.getAppUrls().getStepEditor());
+        setImages(applitoolsVisualCheck.getReadApiKey(), visualCheckResult, stepInfo);
+        return visualCheckResult;
+    }
+
+    private void setImages(String readKey, ApplitoolsVisualCheckResult visualCheckResult, StepInfo stepInfo)
+    {
+        ApiUrls apiUrls = stepInfo.getApiUrls();
+        visualCheckResult.setBaseline(getImageAsBase64(apiUrls.getBaselineImage(), readKey));
+        visualCheckResult.setCheckpoint(getImageAsBase64(apiUrls.getCheckpointImage(), readKey));
+        visualCheckResult.setDiff(getImageAsBase64(apiUrls.getDiffImage(), readKey));
+    }
+
+    private String getImageAsBase64(String url, String readKey)
+    {
+        if (url != null)
+        {
+            URI imageUrl = URI.create(addKeyToUrl(url, readKey));
+            try
+            {
+                HttpResponse response = httpClient.doHttpGet(imageUrl);
```

Comment:
this interaction should be excluded from the Allure report, we don't need Eyes API HTTP request/response -s

Reply:
Since it requires some refactoring -> change aspect based solution to the interceptors based, can we introduce it as a separate PR? I believe this case also not covered in resource checking steps; So it might require even more changes